src/clinical/clinical-api.ts

```diff
@@ -95,6 +95,59 @@ class ClinicalController {
     res.send(zip.toBuffer());
   }
 
+  @HasProgramReadAccess((req: Request) => req.params.programId)
+  async getSpecificClinicalDataAsTsvsInZip(req: Request, res: Response) {
+    const programId: string = req.params.programId;
+    const sort: string = 'donorId';
+    const page: number = 0;
+    const state: CompletionStates = req.query.completionState || CompletionStates.all;
+    const completionState: {} = completionFilters[state] || {};
+
+    const entityTypes: string[] =
+      req.query.entityTypes !== undefined ? req.query.entityTypes.split(',') : [];
+
+    const donorIds: number[] =
+      req.query.donorIds?.match(/\d*/gi)?.filter((match: string) => !!match && parseInt(match)) ||
+      [];
```

Comment:
candidate for reusable utility method

Reply:
`function parseDonorIdList(donorIds: string)=>number[]`

Reply:
Updated, much cleaner